cluster-autoscaler/utils/gce/gce.go

```diff
@@ -164,13 +177,22 @@ func (m *GceManager) GetMigForInstance(instance *config.InstanceConfig) (*config
 	if mig, found := m.migCache[*instance]; found {
 		return mig, nil
 	}
-	if err := m.regenerateCache(); err != nil {
-		return nil, fmt.Errorf("Error while looking for MIG for instance %+v, error: %v", *instance, err)
-	}
-	if mig, found := m.migCache[*instance]; found {
-		return mig, nil
+
+	for _, mig := range m.migs {
+		if mig.config.Project == instance.Project &&
+			mig.config.Zone == instance.Zone &&
+			strings.HasPrefix(instance.Name, mig.basename) {
+			if err := m.regenerateCache(); err != nil {
```

Comment:
I think it'd be readable if you restructure the code such that:

1. in a for loop check if we have any corresponding loop
2. if we have a corresponding mig regenerate cache and return mig
3. if don't have it return nils